packages/pubsub/src/subscription.js

```diff
@@ -592,30 +350,154 @@ Subscription.prototype.decorateMessage_ = function(message) {
  *   var apiResponse = data[0];
  * });
  */
-Subscription.prototype.delete = function(callback) {
+Subscription.prototype.delete = function(gaxOpts, callback) {
   var self = this;
 
-  callback = callback || common.util.noop;
+  if (is.fn(gaxOpts)) {
+    callback = gaxOpts;
+    gaxOpts = {};
+  }
 
-  var protoOpts = {
-    service: 'Subscriber',
-    method: 'deleteSubscription'
+  var reqOpts = {
+    subscription: this.name
   };
 
+  this.request({
+    client: 'subscriberClient',
+    method: 'deleteSubscription',
+    reqOpts: reqOpts,
+    gaxOpts: gaxOpts
+  }, function(err, resp) {
+    if (!err) {
+      self.closeConnection_();
```

Comment:
I'm trying to get a sense of when openConnection_ and closeConnection_ are called.

closeConnection_ calls:
-  when self.messageListeners is decremented to 0
- when deleteSubscription returns with an OK response

openConnection_ calls:
- when a new message listener is added, and the connection is currently closed

There may be others, I haven't read through all of the code yet. But I'm having a difficult time seeing what invariant is being preserved here. For example, we could add 1 message listener, which would call self.openConnection_() as a consequence. We could then add 999 message listeners, none of which would call self.openConnection_(). Then we could call deleteSubscription, which would call self.closeConnection_(). We would then have 1000 message listeners hanging on to a closed connection. Then we could add another message listener, which would call self.openConnection_() again. So we called self.openConnection_() on the 1st and 1001st message listeners. That doesn't make too much sense.